src/games/mapgenoptions.py

```
@@ -93,29 +96,57 @@ def active(self) -> bool:
+        ui_elem.setNoChoiceText(default)
+        super().__init__(name, ui_elem, default, opts)
+
+    def save(self) -> None:
+        config.Settings.set(
+            f"mapGenerator/{self.ui_elem.objectName()}",
+            self.ui_elem.delimiter().join(self.ui_elem.currentData()),
+        )
 
-    def set_value(self, value: int | float) -> None:
+    def as_cmd_arg(self) -> list[str]:
+        return [f"--{self.name}", random.choice(self.ui_elem.currentData())]
```

Ruff (0.15.13)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)